src/Compilers/Core/MSBuildTask/Shared/ManagedCompiler.cs

```diff
@@ -372,8 +372,13 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
                     CompilerServerLogger.Log($"CommandLine = '{commandLineCommands}'");
                     CompilerServerLogger.Log($"BuildResponseFile = '{responseFileCommands}'");
 
+                    var clientDir = TryGetClientDir() ?? Path.GetDirectoryName(pathToTool);
+                    pathToTool = Path.Combine(clientDir, ToolExe);
```

Comment:
It would be good to have a comment why ignoring/overwriting passed in pathToTool is the right thing to do.